### PR TITLE
errors: error handling objects

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: libgiza/error.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+
+class Error(object):
+    """
+    Collects information about, and represents an error encountered. Use in
+    situations where you want to track errors, but don't want to raise
+    exceptions (e.g. for continue on error situations.)
+    """
+
+    def __init__(self, message="generic error", include_trace=True, fatal=True):
```
Comment (i80and) [inline libgiza/error.py]: "generic error" could be pulled out and made a constant, just to make a future typo more detectable.

File: libgiza/error.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+    def dict(self):
+        return {"message": self.message,
+                "payload": self.payload,
+                "fatal": self.fatal,
+                "trace": [{"file": t[0], "line": t[1], "function": t[2], "operation": t[3]}
+                          for t in self.trace]}
+
+    def __repr__(self):
+        # provides a string representation "print(ErrorObject())"
```
Comment (i80and) [inline libgiza/error.py]: It's probably better for this to be a docstring than a comment.

**Comment by tychoish** [comment]
@i80and, assuming the latest run passes, I think this is ready for another pass, (and release shortly thereafter)

**Comment by i80and** [comment]
lgtm!

**Pull request merged**